api/helpers.c

```diff
@@ -653,13 +709,29 @@ void emit_counter64_incr(mambo_context *ctx, void *counter, unsigned incr) {
   emit_a64_LDR_STR_unsigned_immed(ctx, 3, 0, 0, 0, x0, x1);
   emit_a64_pop(ctx, (1 << x0) | (1 << x1));
 #endif
+
+#ifdef __riscv
+  #if __riscv_xlen == 32
+    #error increment 64 not implemented on 32 bit yet
+  #elif __riscv_xlen == 64
+    assert(incr <= 0xFFF);
```

Comment:
It looks like the immediate of addi is sign-extended, so the maximum positive value it can encode is 0x7FF